src/apps/actions.js

```diff
@@ -18,10 +19,23 @@ export function newError(error) {
     };
 }
 
-export function fetchBundleError(bundleId, error) {
+export function fetchBundleError(bundleId, error, bundle) {
     return {
         type: FETCH_BUNDLE_ERROR,
         bundleId,
+        bundle,
         error
     };
 }
+
+export function promulgateBundle(bundle, bundleId, juttleServiceHost, dispatch) {
+    dispatch(newBundle(bundleId, bundle, []));
+
+    return api.describe(juttleServiceHost, bundle)
+    .then(inputs => {
+        dispatch(newBundle(bundleId, bundle, inputs));
```

Comment:
Why do we need to set newBundle twice? Also setting the inputs to `[]` causes the old input values to get cleared, I don't think we should be doing this.